addons/stock/models/stock_quant.py

```diff
@@ -646,6 +646,36 @@ def _compute_complete_name(self):
             res[package.id] = name
         return res
 
+    def _search_location(self, operator, value):
+        if value:
+            packs = self.search([('quant_ids.location_id', operator, value)])
+        else:
+            packs = self.search([('quant_ids', operator, value)])
```

Comment:
In that case you search on packages with quant_ids = False. You don't search on packages with quant_ids.location_id=False and hope receiving something else. The field has one meaning and should stick on it. What is the use case covered by the else clause ?